test/bundler-integration/module/HybridSessionKeyManager.Module.specs.ts

```diff
@@ -0,0 +1,555 @@
+      validUntil,
+      validAfter,
+      sessionKeyData,
+      sessionValidationModule: mockSessionValidationModule1.address,
+    };
+    const sessionData2: ISessionKeyManagerModuleHybrid.SessionDataStruct = {
+      ...sessionData,
+      sessionValidationModule: mockSessionValidationModule2.address,
+    };
+
+    const hybridSKMSingleCallUtils = new HybridSKMSingleCallUtils(
+      entryPoint,
+      hybridSessionKeyManager,
+      ecdsaModule
+    );
+    const hybridSKMBatchCallUtils = new HybridSKMBatchCallUtils(
+      entryPoint,
+      hybridSessionKeyManager,
+      ecdsaModule
+    );
+
+    return {
+      entryPoint: entryPoint,
+      smartAccountImplementation: await getSmartAccountImplementation(),
+      smartAccountFactory: await getSmartAccountFactory(),
+      mockToken: mockToken,
+      ecdsaModule: ecdsaModule,
+      userSA: userSA,
+      hybridSessionKeyManager,
+      mockSessionValidationModule: mockSessionValidationModule1,
+      mockSessionValidationModule2,
+      sessionKeyData: sessionKeyData,
+      sessionData,
+      sessionData2,
+      hybridSKMSingleCallUtils,
+      hybridSKMBatchCallUtils,
+      smartAccountOwner,
+    };
+  });
+
+  describe("Single Call", async () => {
+    it("Should process signed user operation from Session when enabling and use is batched", async () => {
+      const {
+        entryPoint,
+        userSA,
+        hybridSKMSingleCallUtils: utils,
+        mockToken,
+        sessionData,
+        smartAccountOwner,
+      } = await setupTests();
+      const tokenAmountToTransfer = ethers.utils.parseEther("0.834");
+
+      const chainId = (await ethers.provider.getNetwork()).chainId;
+
+      const { sessionEnableData, sessionEnableSignature } =
+        await utils.makeSessionEnableData(
+          [chainId],
+          [sessionData],
+          userSA.address,
+          smartAccountOwner
+        );
+
+      const transferUserOp =
+        await utils.makeEcdsaSessionKeySignedUserOpForEnableAndUseSession(
+          userSA.address,
+          {
+            to: mockToken.address,
+            value: ethers.utils.parseEther("0"),
+            calldata: encodeTransfer(
+              charlie.address,
+              tokenAmountToTransfer.toString()
+            ),
+          },
+          sessionKey,
+          sessionData,
+          sessionEnableData,
+          sessionEnableSignature,
+          0,
+          {
+            preVerificationGas: 51000,
+          }
+        );
+
+      const charlieTokenBalanceBefore = await mockToken.balanceOf(
+        charlie.address
+      );
+
+      await environment.sendUserOperation(transferUserOp, entryPoint.address);
+
+      expect(await mockToken.balanceOf(charlie.address)).to.equal(
+        charlieTokenBalanceBefore.add(tokenAmountToTransfer)
+      );
+    });
+
+    it("Should process signed user operation from Session when session is pre-enabled", async () => {
```

Comment:
- Add negative test cases as well, that will test expired sessions should not allow executing any userOp signed by session key.
- Invalid session key data format, should be reverted.
- Add test cases for disableSession and disableSessions

Reply:
All negative tests are implemented in foundry, please refer to tests starting from
1. https://github.com/bcnmy/scw-contracts/blob/feat/sma-392-session-keys-v2/test/foundry/module/SessionKeyManager/SessionKeyManagerHybrid.BatchCall.t.sol#L526
2. https://github.com/bcnmy/scw-contracts/blob/feat/sma-392-session-keys-v2/test/foundry/module/SessionKeyManager/SessionKeyManagerHybrid.SingleCall.t.sol#L563